### PR TITLE
added alarms and alarm setpoint for co2m

### PR DESCRIPTION
**Pull request by kevin-hereworks**

### CODE REVIEW

**Comment by kevin-hereworks** [comment]
the high_limit_zone_co2_concentration_setpoint is the setpoint at which the alarm will trigger

File: ontology/yaml/resources/fields/telemetry_fields.yaml [hunk 1]
```diff
@@ -3102,3 +3102,7 @@ literals:
 - ultraviolet_irradiance_sensor:
     fixed_min: 0.0
     flexible_max: 2000.0
+
+- high_limit_zone_co2_concentration_setpoint:
```
Comment (shambergoldstein) [inline ontology/yaml/resources/fields/telemetry_fields.yaml:3106]: point already exists as `high_limit_zone_air_co2_concentration_setpoint`

File: ontology/yaml/resources/HVAC/entity_types/ABSTRACT.yaml [hunk 2]
```diff
@@ -1453,6 +1453,8 @@ CO2M:
   - MONITORING
   opt_uses:
   - high_zone_air_co2_concentration_alarm
+  - high_limit_zone_co2_concentration_setpoint
```
Comment (shambergoldstein) [inline ontology/yaml/resources/HVAC/entity_types/ABSTRACT.yaml:1456]: See comment on field below.

File: ontology/yaml/resources/HVAC/entity_types/ABSTRACT.yaml [hunk 3]
```diff
@@ -1453,6 +1453,8 @@ CO2M:
...
+  - failed_alarm
```
Comment (shambergoldstein) [inline ontology/yaml/resources/HVAC/entity_types/ABSTRACT.yaml:1457]: This point would refer to if the general type has failed and should belong on the `SS` for the entity type in most cases. If you are wanting a point to indicate if the co2 sensor has failed then we would need to add that as a new field.

**Review by shambergoldstein — CHANGES_REQUESTED** [review]

File: ontology/yaml/resources/HVAC/entity_types/ABSTRACT.yaml (hunk 3 above)
Comment (kevin-hereworks) [reply ontology/yaml/resources/HVAC/entity_types/ABSTRACT.yaml]: removed thank you

File: ontology/yaml/resources/HVAC/entity_types/ABSTRACT.yaml [hunk 4]
```diff
@@ -1451,13 +1451,13 @@ CO2M:
   guid: "c73f80ff-e57e-4fe4-8bd6-26e4581b7269"
   description: "Basic carbon dioxide monitoring."
   is_abstract: true
+  opt_uses:
+  - high_zone_air_co2_concentration_alarm
+  - high_limit_zone_air_co2_concentration_setpoint
```
Comment (shambergoldstein) [inline ontology/yaml/resources/HVAC/entity_types/ABSTRACT.yaml:1456]: The addition of a setpoint would make this more of a control type than a monitoring type. Would you be able to use `CO2CDSP` for your application and add the alarm as optional there?

**Review by shambergoldstein — CHANGES_REQUESTED** [review]

File: ontology/yaml/resources/HVAC/entity_types/ABSTRACT.yaml (hunk 4 above)
Comment (kevin-hereworks) [reply ontology/yaml/resources/HVAC/entity_types/ABSTRACT.yaml]: yes this could work

**Review by shambergoldstein — APPROVED** [review]

**Pull request merged**